challenge-manager/chain-watcher/watcher.go

```diff
@@ -405,6 +405,34 @@ func (w *Watcher) ComputeAncestors(
 	return chal.honestEdgeTree.ComputeAncestors(ctx, edgeId, blockHeader.Number.Uint64())
 }
 
+func (w *Watcher) IsConfirmableEssentialNode(
+	ctx context.Context,
+	challengedAssertionHash protocol.AssertionHash,
+	essentialNodeId protocol.EdgeId,
+	confirmationThreshold uint64,
+) (bool, []challengetree.EssentialPath, uint64, error) {
+	chal, ok := w.challenges.TryGet(challengedAssertionHash)
+	if !ok {
+		return false, nil, 0, fmt.Errorf("could not get challenge for top level assertion %#x", challengedAssertionHash)
+	}
+	blockHeader, err := w.chain.Backend().HeaderByNumber(ctx, w.chain.GetDesiredRpcHeadBlockNumber())
```

Comment:
Multiple methods within this file use these same 5 lines of code to get the blockHeader and check that the number is a Uint64. Can we extract it into a helper function just to eliminate some repeated code? Maybe even extract the the blockHeader.Number.Uint64() call as well (unless the blockHeader is frequently used in other parts later in the methods.)

Reply:
Done.